src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -576,13 +576,17 @@ private ProcessingBranch PrepareProcessingBranch(Block suggestedBlock, Processin
         ArrayPoolList<Block> blocksToBeAddedToMain = new((int)Reorganization.PersistenceInterval);
 
         bool branchingCondition;
-        bool suggestedBlockIsPostMerge = suggestedBlock.IsPostMerge;
 
         Block toBeProcessed = suggestedBlock;
         long iterations = 0;
         do
         {
             iterations++;
+            if (iterations > MaxBranchSize)
+            {
+                throw new InvalidOperationException($"Maximum size of branch reached ({MaxBranchSize}). This is unexpected.");
```

Comment:
What if we want to do reorg with more blocks? I know that it is currently impossible, but we should aim to be able reorg to the last finalized block and we might have long non-finality events like in Holesky

Reply:
Increase MaxBranchSize?

Reply:
Actually, assuming that a beacon sync was initiated, this should not get called as block downloader would suggest the block one by one.